pkg/apis/hub/v1alpha1/api_auth.go

```diff
@@ -0,0 +1,122 @@
+along with this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// APIAuth defines the authentication configuration for APIs.
+type APIAuth struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired behavior of this APIAuth.
+	Spec APIAuthSpec `json:"spec,omitempty"`
+
+	// The current status of this APIAuth.
+	// +optional
+	Status APIAuthStatus `json:"status,omitempty"`
+}
+
+// APIAuthSpec configures the authentication for APIs.
+// +kubebuilder:validation:XValidation:message="exactly one of apiKey or jwt must be specified",rule="(has(self.apiKey) && !has(self.jwt)) || (!has(self.apiKey) && has(self.jwt))"
+type APIAuthSpec struct {
+	// IsDefault specifies if this APIAuth should be used as the default API authentication method for the namespace.
+	// Only one APIAuth per namespace should have isDefault set to true.
+	IsDefault bool `json:"isDefault"`
+
+	// APIKey configures API key authentication.
+	// +optional
+	APIKey *APIKeyAuthSpec `json:"apiKey,omitempty"`
+
+	// JWT configures JWT authentication.
+	// +optional
+	JWT *JWTAuthSpec `json:"jwt,omitempty"`
+}
+
+// APIKeyAuthSpec configures API key authentication.
+type APIKeyAuthSpec struct{}
+
+// JWTAuthSpec configures JWT authentication.
+// +kubebuilder:validation:XValidation:message="exactly one of signingSecretName, publicKey, jwksFile, or jwksUrl must be specified",rule="[has(self.signingSecretName), has(self.publicKey), has(self.jwksFile), has(self.jwksUrl)].filter(x, x).size() == 1"
+type JWTAuthSpec struct {
+	// StripAuthorizationHeader determines whether to strip the Authorization header before forwarding the request.
+	// +optional
+	StripAuthorizationHeader bool `json:"stripAuthorizationHeader,omitempty"`
+
+	// TokenQueryKey specifies the query parameter name for the JWT token.
+	// +optional
+	TokenQueryKey string `json:"tokenQueryKey,omitempty"`
+
+	// AppIDClaim is the name of the claim holding the identifier of the application.
+	// This field is sometimes named `client_id`.
+	AppIDClaim string `json:"appIdClaim"`
+
+	// TokenNameClaim is the name of the claim holding the name of the token.
+	// This name, if provided, will be used in the metrics.
+	// +optional
+	TokenNameClaim string `json:"tokenNameClaim,omitempty"`
+
+	// ForwardHeaders specifies additional headers to forward with the request.
+	// +optional
+	ForwardHeaders map[string]string `json:"forwardHeaders,omitempty"`
+
+	// SigningSecretName is the name of the Kubernetes Secret containing the signing secret.
+	// The secret must be of type Opaque and contain a key named 'value'.
+	// +optional
+	// +kubebuilder:validation:MaxLength=253
+	SigningSecretName string `json:"signingSecretName,omitempty"`
```

Comment:
`base64encoded` attribute is missing.

Reply:
I double checked, the `Base64Encoded` option that we provide in the middleware is only here to tell if the signing secret is passed base64 encoded or not. Since we are on a secret, the value will be base64 decoded automatically.